sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -301,12 +301,24 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
   def start(): StreamingQuery = startInternal(None)
 
   /**
+   * :: Experimental ::
+   *
    * Starts the execution of the streaming query, which will continually output results to the given
    * table as new data arrives. A new table will be created if the table not exists. The returned
```

Comment:
Here it documented a new table will be created if not existing, but later it also documents "Please create a table manually before the execution". It looks confusing, I think. Could we rephrase them together and give a more concrete description about table creation?

Reply:
Maybe we have two small paragraphs for v1 and v2 table separately? E.g.

```
For v1 table, partitioning columns provided by `partitionBy` will be respected no matter the table exists or not. A new table will be created if the table not exists.

For v2 table, `partitionBy` will be ignored if the table already exists. `partitionBy` will be respected only if the v2 table does not exist. Besides, the v2 table created by this API lacks some functionalities (e.g., customized properties, options, and serde info). If you need them, please create the v2 table manually before the execution to avoid creating a table with incomplete information.
```

Reply:
+1 to @viirya suggestion. My request was to describe the impact of options (mostly partitionBy) for matrix of v1 vs v2 and existing vs non-existing. 

Separating the case of v1 vs v2 is more important, so the suggestion looks better.

Reply:
Thanks for the rephrase, done in c158775